homeassistant/components/vesync/sensor.py

```diff
@@ -0,0 +1,152 @@
+"""Support for power & energy sensors for VeSync outlets."""
+from datetime import datetime
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.const import ENERGY_KILO_WATT_HOUR, POWER_WATT
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import EntityCategory
+
+from .common import VeSyncBaseEntity
+from .const import DOMAIN, VS_DISCOVERY, VS_DISPATCHERS, VS_SENSORS
+from .switch import DEV_TYPE_TO_HA
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up switches."""
+
+    async def async_discover(devices):
```

Comment:
This doesn't need to be a coroutine function since we don't await inside. Make it a normal function and decorate it with `@callback` imported from core.py.